HeterogeneousCore/CUDAServices/interface/CUDAService.h

```diff
@@ -0,0 +1,46 @@
+#ifndef HeterogeneousCore_CUDAServices_CUDAService_h
+#define HeterogeneousCore_CUDAServices_CUDAService_h
+
+#include <utility>
+#include <vector>
+
+#include "FWCore/Utilities/interface/StreamID.h"
+
+namespace edm {
+  class ParameterSet;
+  class ActivityRegistry;
+  class ConfigurationDescriptions;
+}  // namespace edm
+
+/**
+ * TODO:
+ * - CUDA stream management?
+ *   * Not really needed until we want to pass CUDA stream objects from one module to another
+ *   * Which is not really needed until we want to go for "streaming mode"
+ *   * Until that framework's inter-module synchronization is safe (but not necessarily optimal)
+ * - Management of (preallocated) memory?
+ */
+class CUDAService {
+public:
+  CUDAService(edm::ParameterSet const& iConfig, edm::ActivityRegistry& iRegistry);
```

Comment:
Since you do not use the ActivityRegistry, it is possible to create a service without that parameter. See

https://github.com/cms-sw/cmssw/blob/master/FWCore/Integration/test/ValueExamplePlugin.cc#L19

Reply:
Thanks for pointing it out, will remove.